website/static/js/projectSettings.js

```diff
@@ -107,7 +107,40 @@ var ProjectSettings = oop.extend(
                     }
                 }
             };
-        }
+        },
+        enableMailingList: function() {
+            var request = $.ajax({
+                type: 'post',
+                dataType: 'json',
+                url: nodeApiUrl + 'mailing_list/'
+            });
+            request.done(function() {
+                window.location.reload();
+            });
+            request.fail($osf.handleJSONError);
+        },
+        disableMailingList: function(nodeType) {
+
+            var message = '<p>Are you sure that you want to disable the mailing list for this ' + nodeType + '?';
```

Comment:
Gosh. `project-settings-page.js` [includes](https://github.com/CenterForOpenScience/osf.io/blob/af226bba8d0f9159151aafcdb0e69da4cd9d67d5/website/static/js/pages/project-settings-page.js#L8) `projectSettings.js`. So the latter is abstract, concretized in the former?